src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.NtAuth.cs

```diff
@@ -77,7 +78,23 @@ private static async Task<HttpResponseMessage> SendWithNtAuthAsync(HttpRequestMe
 
                         string challengeData = challenge.ChallengeData;
 
-                        string spn = "HTTP/" + authUri.IdnHost;
+                        // Need to use FQDN normalized host so that CNAME's are traversed.
+                        string spn;
+                        if (authUri.HostNameType == UriHostNameType.IPv6 || authUri.HostNameType == UriHostNameType.IPv4)
```

Comment:
Nit: can you store HostNameType into a local so that in the common case we only need to access it once rather than twice?  It looks like it caches but even then that it's not super-cheap or inlineable.

Reply:
Will fix.